qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/common/cropping/cropping_presenter.py

```diff
@@ -19,110 +19,84 @@ def __init__(self, parent, view=None, model=None):
-        self.custom_calfile = None
-        self.calfile_valid = True
+        self.custom_groupingfile_enabled = True
+        self.custom_groupingfile = None
+        self.groupingfile_valid = True
 
         # Connect view signals to local functions
         self.view.set_on_combo_changed(self.on_combo_changed)
-        self.view.set_on_custom_calfile_changed(self.on_calfile_changed)
+        self.view.set_on_custom_groupingfile_changed(self.on_groupingfile_changed)
         self.view.set_on_custom_spectra_changed(self.on_spectra_changed)
 
         self.on_combo_changed(0)
 
     # Signal Activated Functions
 
-    def on_combo_changed(self, index):
-        if index == 0:  # custom calfile
-            self.group = GROUP.CUSTOM
-            self.custom_calfile_enabled = True
-            self.custom_spectra_enabled = False
-            self.set_custom_widgets_visibility(True, False)
-        elif index == 1:  # north
-            self.group = GROUP.NORTH
-            self.custom_calfile_enabled = False
-            self.custom_spectra_enabled = False
-            self.set_custom_widgets_visibility(False, False)
-        elif index == 2:  # south
-            self.group = GROUP.SOUTH
-            self.custom_calfile_enabled = False
-            self.custom_spectra_enabled = False
-            self.set_custom_widgets_visibility(False, False)
-        elif index == 3:  # cropped
-            self.group = GROUP.CROPPED
-            self.custom_calfile_enabled = False
-            self.custom_spectra_enabled = True
-            self.set_custom_widgets_visibility(False, True)
-        elif index == 4:  # texture 20 grouping
-            self.group = GROUP.TEXTURE20
-            self.custom_calfile_enabled = False
-            self.custom_spectra_enabled = False
-            self.set_custom_widgets_visibility(False, False)
-        else:
-            self.group = GROUP.TEXTURE30
-            self.custom_calfile_enabled = False
-            self.custom_spectra_enabled = False
-            self.set_custom_widgets_visibility(False, False)
-
-    def on_calfile_changed(self):
-        valid = self.view.finder_custom.isValid()
-        self.calfile_valid = valid
-        if valid:
-            self.custom_calfile = self.view.get_custom_calfile()
-        else:
-            self.custom_calfile = None
+    def on_combo_changed(self, index: int) -> None:
+        """
+        Handles the event when the combo box selection is changed.
+        GROUP.TEXTURE30 is the default group if no cropping is requested.
+
+        :param index: The index of the selected item in the combo box.
+        """
+        group_mapping = {0: GROUP.CUSTOM, 1: GROUP.NORTH, 2: GROUP.SOUTH, 3: GROUP.CROPPED, 4: GROUP.TEXTURE20, 5: GROUP.TEXTURE30}
+        self.group = group_mapping.get(index, GROUP.TEXTURE30)
+        self.custom_groupingfile_enabled = index == 0
+        self.custom_spectra_enabled = index == 3
+        self.set_custom_widgets_visibility(index == 0, index == 3)
```

Comment:
Not sure if it's more readable but technically avoids some small duplication
```suggestion
        self.set_custom_widgets_visibility(self.custom_groupingfile_enabled, self.custom_spectra_enabled)
```